lib/ansible/modules/system/known_hosts.py

```diff
@@ -150,11 +156,11 @@ def enforce_state(module, params):
             module.atomic_move(outf.name, path)
         except (IOError, OSError) as e:
             module.fail_json(msg="Failed to write to file %s: %s" % (path, to_native(e)))
-
-        try:
-            outf.close()
-        except:
-            pass
+        finally:
```

Comment:
`tempfile.NamedTemporaryFile` returns a file like-object which can be used in a with statement ([example](https://docs.python.org/3/library/tempfile.html#examples)). This will allow to call `close` automatically, then the finally block could be removed.

Reply:
@pilou- Which versions of python need to be supported on the remote node? I am aware of how `with` works but didn't change it  because it wasn't in use…